### PR TITLE
Spring 01

### PR DESCRIPTION
**Pull request by Rudiachenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,90 @@
...
+            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
 
+<dependencies>
+    <dependency>
+        <groupId>org.hibernate</groupId>
+        <artifactId>hibernate-java8</artifactId>
+        <version>5.4.21.Final</version>
+    </dependency>
```
Comment (jah701) [inline pom.xml:26]: Do we need this dependency?

File: pom.xml [hunk 2]
```diff
@@ -8,5 +8,90 @@
...
+                    <version>3.8.0</version>
+                    <configuration>
+                        <source>11</source>
+                        <target>11</target>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 </project>
```
Comment (jah701) [inline pom.xml]: Empty line

File: spring.iml [hunk 3]
```diff
@@ -1,2 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<module type="JAVA_MODULE" version="4" />
```
Comment (jah701) [inline spring.iml:2]: Do we need this .iml file in repository?

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert user " + user.toString(), e);
```
Comment (jah701) [inline src/main/java/dao/impl/UserDaoImpl.java]: User's method "toString()" will be invoked automatically

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<User> listUsers() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM User", User.class).getResultList();
```
Comment (jah701) [inline src/main/java/dao/impl/UserDaoImpl.java:43]: Would you like to use CriteriaAPI for select queries as it is more efficient?

Comment (Rudiachenko) [reply src/main/java/dao/impl/UserDaoImpl.java:43]: Why CriteriaAPI  is more efficient?

File: README.md [hunk 6]
```diff
@@ -1 +1,22 @@
 "# spring" 
++ Create a new project and link it to the GitHub
```
Comment (RomanDubovskyi) [inline README.md]: let's remove task description from readme

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**